src/Passes/Source/Rules/Factory.hpp

```diff
@@ -69,6 +69,7 @@ namespace quantum
         /// @{
         void optimiseBranchQuantumOne();
         void optimiseBranchQuantumZero();
+        void optimiseSelectQuantumOne();
         /// @}
```

Comment:
Nit:
The mismatch between the file name (Factory.hpp) and the class name (**Rule**Factory), defined in the that file, is suspicious.